src/atoms.ts

```diff
@@ -2,4 +2,11 @@ import { atomWithMMKV } from "./atomWithMMKV";
 
 export const apiKeyAtom = atomWithMMKV("apiKey", "");
 export const serverAtom = atomWithMMKV("server", "");
+
+export const lnbitsUrlAtom = atomWithMMKV("lnbitsUrl", "");
```

Comment:
Consider combining all of those values into one value. Since domain user and wallet is always extracted from lnbitsUrl.


put this into utils/parseLnbitsUrl.ts
```typescript
export default function parseLnbitsUrl(url: string) {
     const fullUrl = new URLParse(url, true);

     const domain = fullUrl.origin;
     const user = fullUrl.query.usr;
     const wallet = fullUrl.query.wal;

     return { domain, user, wallet };
   }
```

```suggestion

interface UserInfo {
  domain: string
  user: string
  wallet: string
}

export const lnbitsUrlAtom = atomWithMMKV<string | null>("lnbitsurl", null)

export const userInfoAtom = selectAtom(lnbitsUrlAtom, (url) => {
  if(!url) return null
  return parseLnbitsUrl(url)
})
```